integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,502 @@
+            condition -> logger.info("Waiting for domain {0} to be created in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                domainUid,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainExists(domainUid, DOMAIN_VERSION, domainNamespace));
+
+    String cloudProvider = "baremetal";
+    boolean enableValidatingWebhook = false;
+    voyagerHelmParams = installAndVerifyVoyager(voyagerNamespace,
+        cloudProvider, enableValidatingWebhook);
+  }
+
+  /**
+   * Verify all server pods are running.
+   * Verify all k8s services for all servers are created.
+   */
+  @BeforeEach
+  public void beforeEach() {
+    logger.info("Check admin service and pod {0} is created in namespace {1}",
+        adminServerPodName, domainNamespace);
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace);
+    // check managed server services and pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server services and pods are created in namespace {0}",
+          domainNamespace);
+      checkPodReadyAndServiceExists(managedServerPrefix + i, domainUid, domainNamespace);
+    }
+  }
+
+  @Order(1)
+  @Test
+  @DisplayName("Verify the RMI access WLS through LoadBalancer tunneling port")
+  public void testExternalRmiAccessThruHttpTunneling() {
+
+    buildClient();
+
+    // Prepare the voyager ingress file from the template file by replacing 
+    // domain namespace, domain UID, cluster service name and tls secret 
+    Map<String, String> templateMap  = new HashMap();
+    templateMap.put("DOMAIN_NS", domainNamespace);
+    templateMap.put("DOMAIN_UID", domainUid);
+    templateMap.put("CLUSTER", clusterName);
+
+    Path srcVoyagerHttpFile = Paths.get(RESOURCE_DIR, 
+        "tunneling", "voyager.tunneling.template.yaml");
+    Path targetVoyagerHttpFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcVoyagerHttpFile.toString(), 
+        "voyager.tunneling.yaml", templateMap));
+    logger.info("Generated Voyager Http Tunneling file {0}", targetVoyagerHttpFile);
+
+    StringBuffer deployIngress = new StringBuffer("kubectl apply -f ");
+    deployIngress.append(Paths.get(RESULTS_ROOT, "voyager.tunneling.yaml"));
+
+    ExecResult result = assertDoesNotThrow(
+        () -> exec(new String(deployIngress), true));
+
+    logger.info("kubectl apply returned {0}", result.toString());
+    checkServiceExists("voyager-voyager-tunneling", domainNamespace);
+
+    int httpTunnelingPort =
+        getServiceNodePort(domainNamespace, "voyager-voyager-tunneling");
+    assertTrue(httpTunnelingPort != -1,
+        "Could not get the HttpTunnelingPort service node port");
+    logger.info("HttpTunnelingPort for Voyager {0}", httpTunnelingPort);
+
+    StringBuffer httpUrl = new StringBuffer("http://");
```

Comment:
comments

Reply:
// Generate java command to execute client with classpath